### ISSUE TITLE
Remove `scikit-learn` dependency

### ISSUE THREAD

**Issue by dylanbouchard** [cvs-health/uqlm#3]
**Is your feature request related to a problem? Please describe.**
Currently, we use `sklearn` for classification metrics, including precision, recall, f-score, roc-auc, and log-loss. We would like to remove the dependency and write these simple metrics ourselves.

**Describe the solution you'd like**
Write these metrics as utility functions using `numpy`

**Describe alternatives you've considered**
status quo

**Comment by dylanbouchard** [comment]
This issue is assigned to @mohitcek

**Comment by mohitcek** [comment]
> This issue is assigned to [@mohitcek](https://github.com/mohitcek)

Thank you @dylanbouchard, I will get back to you after adding these metrics to the utils module.

**Comment by dylanbouchard** [comment]
Closing this issue as we have decided not to pursue this as other planned features will leverage `scikit-learn`.
